pkg/apis/core/validation/validation.go

```diff
@@ -2920,6 +2928,18 @@ func ValidateVolumeMounts(mounts []core.VolumeMount, voldevices map[string]strin
 			allErrs = append(allErrs, field.Invalid(idxPath.Child("mountPath"), mnt.MountPath, "must not already exist as a path in volumeDevices"))
 		}
 
+		// Disallow subPath/subPathExpr for image volumes
+		if opts.AllowImageVolumeSource {
+			if v, ok := volumes[mnt.Name]; ok && v.Image != nil {
+				if len(mnt.SubPath) != 0 {
+					allErrs = append(allErrs, field.Invalid(idxPath.Child("subPath"), mnt.SubPath, "not allowed in image volume sources"))
```

Comment:
IMO - it should be specified in api docs that, using `subpath` for this volume type is not permitted.

Reply:
Is this the line you're looking for? https://github.com/kubernetes/kubernetes/blob/0caeba5cbe7793d09586556e056b08f4a8e0adba/pkg/apis/core/types.go#L199

Reply:
I would have thought, it will be also documented in `VolumeMounts` field doc.

Reply:
Hm, I don't think that `VolumeMount` is the right place to document that. :thinking: